src/main/java/de/dennisguse/opentracks/content/provider/ContentProviderUtils.java

```diff
@@ -745,6 +745,16 @@ public TrackPointIterator getTrackPointLocationIterator(final Track.Id trackId,
         return new TrackPointIterator(this, trackId, startTrackPointId);
     }
 
+    public List<TrackPoint> getTrackPoints(final Track.Id trackId, final TrackPoint.Id startTrackPointId) {
+        List<TrackPoint> trackPoints = new ArrayList<>();
+        try (TrackPointIterator trackPointIterator = getTrackPointLocationIterator(trackId, startTrackPointId)) {
+            while (trackPointIterator.hasNext()) {
+                trackPoints.add(trackPointIterator.next());
```

Comment:
there might be a nicer way: https://www.baeldung.com/java-convert-iterator-to-list

(Nr. 3 or Nr. 4)